lib/service_actor/checks/type_check.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+# Adds `type:` checking to inputs and outputs. Accepts class names or classes
+# that should match an ancestor. Also accepts arrays.
+#
+# Example:
+#
+#   class ReduceOrderAmount < Actor
+#     input :order, type: "Order"
+#     input :amount, type: [Integer, Float]
+#     input :bonus_applied, type: [TrueClass, FalseClass]
+#   end
+#
+#   class ReduceOrderAmount < Actor
+#     input :order, type: { is: Order, message: "Order is required" }
+#     input :amount, type: { is: Integer, message: "Incorrect amount" }
+#
+#     input :bonus_applied,
+#           type: {
+#             is: [TrueClass, FalseClass],
+#             message: (lambda do |origin:, input_key:, actor:, expected_type:, given_type:| # rubocop:disable Layout/LineLength
+#               "Wrong type \"#{given_type}\" for \"#{input_key}\". " \
+#               "Expected: \"#{expected_type}\""
+#             end)
+#           }
+#   end
+class ServiceActor::Checks::TypeCheck < ServiceActor::Checks::Base
+  DEFAULT_MESSAGE = lambda do
+    |origin:, input_key:, actor:, expected_type:, given_type:|
+
+    "The \"#{input_key}\" #{origin} on \"#{actor}\" must be of type " \
+      "\"#{expected_type}\" but was \"#{given_type}\""
+  end
+
+  private_constant :DEFAULT_MESSAGE
+
+  def self.check( # rubocop:disable Metrics/ParameterLists
+    check_name:,
+    origin:,
+    input_key:,
+    actor:,
+    conditions:,
+    result:,
+    **
+  ) # do
+    return unless check_name == :type
+
+    new(
+      origin: origin,
+      input_key: input_key,
+      actor: actor,
+      type_definition: conditions,
+      given_type: result[input_key],
```

Comment:
Should we give the actual class object instead of the value?
```suggestion
      given_type: result[input_key].class,
```

Reply:
It breaks behavior. This is not the purpose of this PR.